spinnaker/spinnaker_system/google_http_upsert_server_test.py

```diff
@@ -0,0 +1,195 @@
+import citest.service_testing as st
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+
+
+SCOPES = [gcp.COMPUTE_READ_WRITE_SCOPE]
+
+class GoogleHttpLoadBalancerTest(st.AgentTestCase):
+  '''Test fixture for Http LB test.
+  '''
+
+  FIRST_CERT = ''
+  SECOND_CERT = ''
+
+  @classmethod
+  def setUpClass(cls):
+    cls.FIRST_CERT = 'first-cert' + str(uuid.uuid4())
+    cls.SECOND_CERT = 'second-cert' + str(uuid.uuid4())
+    runner = citest.base.TestRunner.global_runner()
+    bindings = runner.bindings
+    scenario = runner.get_shared_data(upsert.GoogleHttpLoadBalancerTestScenario)
+    managed_region = runner.bindings['TEST_GCE_REGION']
+    title = 'Check Quota for {0}'.format(scenario.__class__.__name__)
+
+    verify_results = gcp.verify_quota(
+      title,
+      scenario.gcp_observer,
+      project_quota=upsert.GoogleHttpLoadBalancerTestScenario.MINIMUM_PROJECT_QUOTA,
+      regions=[(managed_region,
+                upsert.GoogleHttpLoadBalancerTestScenario.MINIMUM_REGION_QUOTA)]
+    )
+    if not verify_results:
+      raise RuntimeError('Insufficient Quota: {0}'.format(verify_results))
+
+    key = crypto.PKey()
+    key.generate_key(crypto.TYPE_RSA, 2048)
+    cert = crypto.X509()
+    cert.get_subject().C = "US"
+    cert.get_subject().ST = "New York"
+    cert.get_subject().L = "New York City"
+    cert.get_subject().O = "Google"
+    cert.get_subject().OU = "Spinnaker"
+    cert.get_subject().CN = "localhost"
+    cert.set_serial_number(4096)
+    cert.gmtime_adj_notBefore(0)
+    cert.gmtime_adj_notAfter(24 * 60 * 60)
+    cert.set_issuer(cert.get_subject())
+    cert.set_pubkey(key)
+    cert.sign(key, 'sha1')
+
+    compute_agent = gcp.GcpComputeAgent.make_agent(scopes=SCOPES,
+                                                   credentials_path=bindings['GCE_CREDENTIALS_PATH'])
+    context = ExecutionContext()
+    ssl_cert = {
+      'name': cls.FIRST_CERT,
+      'privateKey': crypto.dump_privatekey(crypto.FILETYPE_PEM, key),
+      'certificate': crypto.dump_certificate(crypto.FILETYPE_PEM, cert)
+    }
+    compute_agent.invoke_resource(context, # No predicates against this agent, context is empty.
+                                  'insert',
+                                  resource_type='sslCertificates',
+                                  project=bindings['GCE_PROJECT'],
+                                  body=ssl_cert)
+    ssl_cert['name'] = cls.SECOND_CERT
+    compute_agent.invoke_resource(context,
+                                  'insert',
+                                  resource_type='sslCertificates',
+                                  project=bindings['GCE_PROJECT'],
+                                  body=ssl_cert)
+
+
+  @classmethod
+  def tearDownClass(cls):
+    runner = citest.base.TestRunner.global_runner()
+    bindings = runner.bindings
+    context = ExecutionContext()
+    compute_agent = gcp.GcpComputeAgent.make_agent(scopes=SCOPES,
```

Comment:
Seems like a bit of overhead that accomplishes the same thing as having the `tearDownClass`. What is the difference between queuing functions to run and having an explicit teardown block and running the same functions you would queue?

The `context`'s utility would be the determining factor here, otherwise it's very similar to what's available.